code/modules/projectiles/guns/flamer.dm

```diff
@@ -330,10 +330,8 @@
 	starting_attachment_types = list(/obj/item/attachable/flamer_nozzle, /obj/item/attachable/stock/t84stock)
 	var/obj/item/storage/holster/backholster/flamer/flamerpack
 
-/obj/item/weapon/gun/flamer/big_flamer/marinestandard/engineer/removed_from_inventory(mob/user)
-	. = ..()
-	flamerpack.handle_item_insertion(src)
-	flamerpack.refuel(chamber_items[1], user)
+/obj/item/weapon/gun/flamer/big_flamer/marinestandard/engineer/on_enter_storage(obj/item/storage/S)
+	flamerpack.refuel(chamber_items[1], loc.loc)
```

Comment:
?? this is very snowflake. The var exists on the gun, but is never actually assigned anything by the gun itself, it relies on the backpack's code, but this special reload proc is on the gun?

This will also runtime if there is no flamerpack, and there is no check if you're even storing it in a flamerpack (or the right one) to begin with. All this stuff should be handled by the backpack itself.

Reply:
is what I now fine?